### PR TITLE
Fixing point cloud transformation bug with using uint16 when int16 sh…

### LINKED ISSUE #1556: Python API doesn't transform correctly to point cloud from depth image
**Describe the bug**

Python API doesn't transform correctly to point cloud from depth image.
The API returns a result, but it does not seem to be the correct point cloud.

**To Reproduce**

1. run this simple script for save point cloud to file
    ```py
    import numpy as np
    import k4a
    
    if __name__ == '__main__':
        # Open Device
        device = k4a.Device.open()
    
        # Start Cameras
        device_config = k4a.DEVICE_CONFIG_BGRA32_1080P_NFOV_UNBINNED_FPS15
        device.start_cameras(device_config)

        # Get Calibration
        calibration = device.get_calibration(
            depth_mode=device_config.depth_mode,
            color_resolution=device_config.color_resolution)

        # Create Transformation
        transformation = k4a.Transformation(calibration)

        # Capture One Frame
        capture = device.get_capture(-1)

        # Get Point Cloud
        xyz_image = transformation.depth_image_to_point_cloud(capture.depth, k4a.ECalibrationType.DEPTH)
    
        # Save Point Cloud To Ascii Format File
        height, width, channels = xyz_image.data.shape
        xyz_data = xyz_image.data.reshape(height * width, channels) # convert to (1080 * 1920, 3) from (1080, 1920, 3)
        np.savetxt('data.txt', xyz_data, delimiter=' ', fmt='%u') # save ascii format (x y z\n x y z\n x y z\n ...)
    
        # Stop Cameras
        device.stop_cameras()
    ```
2. open ascii format point cloud data using meshlab
    2021-04-02_16h01_18

**Expected behavior**

It can get correctly transformed point cloud data. 

**Logs**

**Screenshots**

**Desktop (please complete the following information):**
 - OS with Version: Windows 10 20H2 
 - SDK Version:  develop/HEAD 
 - Python Version: Python 3.8
 - NumPy Version: NumPy 1.20.0

**Additional context**

The other features of Python API are working correctly.

### PR DESCRIPTION
**Pull request by JonathanESantos**
### Description of the changes:
- Fixing point cloud transformation bug with using uint16 when int16 should be used because the X and Y images are centered around 0 and can take on negative values.
- Adding point_cloud_capture.py example to capture depth data, transform to point cloud, and write to text file which can then be opened with a 3D modeling application like MeshLab.
-

### Before submitting a Pull Request:
- [x] I reviewed [CONTRIBUTING.md](https://github.com/Microsoft/Azure-Kinect-Sensor-SDK/blob/develop/CONTRIBUTING.md)
- [x] I [built my changes](https://github.com/Microsoft/Azure-Kinect-Sensor-SDK/blob/develop/docs/building.md) locally
- [x] I ran the [unit tests](https://github.com/Microsoft/Azure-Kinect-Sensor-SDK/blob/develop/docs/testing.md)
- [x] I ran the [functional tests](https://github.com/Microsoft/Azure-Kinect-Sensor-SDK/blob/develop/docs/testing.md) with a device
- [x] I ran the [performance tests](https://github.com/Microsoft/Azure-Kinect-Sensor-SDK/blob/develop/docs/testing.md) with a device

### I tested changes on: 
- [x] Windows
- [ ] Linux

- [x] Python automated unit tests
- [x] Python automated functional_fast tests.

### CODE REVIEW

**Review by qm13 — DISMISSED** [review]

**Review by jaygullapalli — COMMENTED** [review]
/azp run

**Comment by UnaNancyOwen** [comment]
It works fine! Thanks, 👍 

@JoachimHafner I have one question. (I'm newbie about python/numpy.)
These two conversion methods look like same processing result. 
Why did you choose former method?
```py
# point_cloud_capture.py#L27-L31
width, height, channels = xyz_image.data.shape
xyz_data = np.empty((width * height, channels,), dtype=xyz_image.data.dtype)
xyz_data[:,0] = xyz_image.data[:,:,0].reshape(width*height)
xyz_data[:,1] = xyz_image.data[:,:,1].reshape(width*height)
xyz_data[:,2] = xyz_image.data[:,:,2].reshape(width*height)
```
```py
height, width, channels = xyz_image.data.shape
xyz_data = np.reshape(xyz_image.data, (height * width, channels))
```

**Comment by UnaNancyOwen** [comment]
It works fine! Thanks, 👍 

@JonathanESantos I have one question. (I'm newbie about python/numpy.)
These two reshape methods look like same processing result. 
Why did you choose former method?
```py
# point_cloud_capture.py#L27-L31
width, height, channels = xyz_image.data.shape
xyz_data = np.empty((width * height, channels,), dtype=xyz_image.data.dtype)
xyz_data[:,0] = xyz_image.data[:,:,0].reshape(width*height)
xyz_data[:,1] = xyz_image.data[:,:,1].reshape(width*height)
xyz_data[:,2] = xyz_image.data[:,:,2].reshape(width*height)
```
```py
height, width, channels = xyz_image.data.shape
xyz_data = xyz_image.data.reshape(height * width, channels)
```

**Comment by JonathanESantos** [comment]
> It works fine! Thanks, 👍
> 
> @JonathanESantos I have one question. (I'm newbie about python/numpy.)
> These two reshape methods look like same processing result.
> Why did you choose former method?
> 
> ```python
> # point_cloud_capture.py#L27-L31
> width, height, channels = xyz_image.data.shape
> xyz_data = np.empty((width * height, channels,), dtype=xyz_image.data.dtype)
> xyz_data[:,0] = xyz_image.data[:,:,0].reshape(width*height)
> xyz_data[:,1] = xyz_image.data[:,:,1].reshape(width*height)
> xyz_data[:,2] = xyz_image.data[:,:,2].reshape(width*height)
> ```
> 
> ```python
> height, width, channels = xyz_image.data.shape
> xyz_data = xyz_image.data.reshape(height * width, channels)
> ```

@UnaNancyOwen I'm also not very well-versed in numpy, but they should be the same result, but I think the reshape does not create a copy of the data whereas the former does. I don't know remember why I chose the former over the latter. I do remember having to fix the function to return the XYZ in planes rather than interleaved, and maybe I was doing some debugging and explicitly splicing the data and viewing it to check the results, and just sort of left it looking like that.

For speed, I would definitely go with the reshape since it does not create a copy of the data. I'll modify it to use the reshape and check the results. Thanks for checking!

File: src/python/k4a/examples/point_cloud_capture.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    transformation = k4a.Transformation(calibration)
+
+    # Capture One Frame
+    capture = device.get_capture(-1)
+
+    # Get Point Cloud
+    xyz_image = transformation.depth_image_to_point_cloud(capture.depth, k4a.ECalibrationType.DEPTH)
+
+    # Save Point Cloud To Ascii Format File. Interleave the [X, Y, Z] channels into [x0, y0, z0, x1, y1, z1, ...]
+    width, height, channels = xyz_image.data.shape
```
Comment (UnaNancyOwen) [inline src/python/k4a/examples/point_cloud_capture.py]: It is wrong order of shape. The correct order is height, width, and channels. Please swap width and height.
```diff
- width, height, channels = xyz_image.data.shape
+ height, width, channels = xyz_image.data.shape
```

File: src/python/k4a/examples/point_cloud_capture.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    xyz_data = np.empty((width * height, channels,), dtype=xyz_image.data.dtype)
```
Comment (UnaNancyOwen) [inline src/python/k4a/examples/point_cloud_capture.py]: I think this line is not required. Please consider to remove this line to keep simple.
```diff
- xyz_data = np.empty((width * height, channels,), dtype=xyz_image.data.dtype)
```

File: src/python/k4a/examples/point_cloud_capture.py (hunk 1 above)
Comment (JonathanESantos) [reply src/python/k4a/examples/point_cloud_capture.py]: Fixed with most recent update.

File: src/python/k4a/examples/point_cloud_capture.py (hunk 2 above)
Comment (JonathanESantos) [reply src/python/k4a/examples/point_cloud_capture.py]: Removed with most recent update.

**Review by jaygullapalli — APPROVED** [review]

**Pull request merged**